Framework/API/src/WorkspaceUnitValidator.cpp

```diff
@@ -38,19 +53,39 @@ WorkspaceUnitValidator::checkValidity(const MatrixWorkspace_sptr &value) const {
   Kernel::Unit_const_sptr unit = value->getAxis(0)->unit();
   // If m_unitID is empty it means that the workspace must have units, which
   // can be anything
-  if (m_unitID.empty()) {
+  if (m_unitIDs.empty()) {
     return (
         unit && (!boost::dynamic_pointer_cast<const Kernel::Units::Empty>(unit))
             ? ""
             : "The workspace must have units");
   }
   // now check if the units of the workspace is correct
   else {
-    if ((!unit) || (unit->unitID().compare(m_unitID))) {
-      return "The workspace must have units of " +
-             m_unitID; //+ "; its unit is: " + unit->caption();
-    } else
-      return "";
+    const auto matchesUnitID =
+        [&unit](const std::string &unitID) { return unit->unitID() == unitID; };
+
+    if (!unit ||
+        !std::any_of(m_unitIDs.cbegin(), m_unitIDs.cend(), matchesUnitID)) {
```

Comment:
I find NOT OR operations slightly unintuitive to see in the code clearly, a comment here might help.